### PR TITLE
move callback invocation to completionHandler of DismissViewController

### PR DESCRIPTION
**Pull request by jennyf19**
**Manually tested with:**
>ADAL iOS test app on simulator
>MSAL iOS test app on simulator w/embedded webview

#1016

### CODE REVIEW

File: core/src/Platforms/iOS/EmbeddedWebview/WKWebNavigationDelegate.cs [hunk 1]
```diff
@@ -73,7 +73,8 @@ public override void DecidePolicy(WKWebView webView, WKNavigationAction navigati
                    requestUrlString.StartsWith(BrokerConstants.BrowserExtInstallPrefix, StringComparison.OrdinalIgnoreCase))
             {
                 AuthenticationAgentUIViewController.callbackMethod(new AuthorizationResult(AuthorizationStatus.Success, requestUrlString));
```
Comment (henrik-me) [inline core/src/Platforms/iOS/EmbeddedWebview/WKWebNavigationDelegate.cs]: >AuthenticationAgentUIViewController.callbackMethod(new AuthorizationResult(AuthorizationStatus.Success, requestUrlString)); [](start = 16, length = 123)

remove ? Seems like you have two calls to callBackMethod #Resolved

File: core/src/Platforms/iOS/EmbeddedWebview/WKWebNavigationDelegate.cs [hunk 2]
```diff
@@ -73,7 +73,8 @@ public override void DecidePolicy(WKWebView webView, WKNavigationAction navigati
...
-                AuthenticationAgentUIViewController.DismissViewController(true, null);
+                AuthenticationAgentUIViewController.DismissViewController(true, () =>
+                    AuthenticationAgentUIViewController.callbackMethod(new AuthorizationResult(AuthorizationStatus.Success, requestUrlString)));
                 decisionHandler(WKNavigationActionPolicy.Cancel);
```
Comment (henrik-me) [inline core/src/Platforms/iOS/EmbeddedWebview/WKWebNavigationDelegate.cs]: >Cancel [](start = 57, length = 6)

Should this be cancel and not Ok/Success or something similar? #Resolved

**Review by henrik-me — CHANGES_REQUESTED** [review]
:clock1:

File: core/src/Platforms/iOS/EmbeddedWebview/WKWebNavigationDelegate.cs (hunk 1 above)
Comment (jennyf19) [reply core/src/Platforms/iOS/EmbeddedWebview/WKWebNavigationDelegate.cs]: done...thanks. Forgot to remove that line.  #Resolved

File: core/src/Platforms/iOS/EmbeddedWebview/WKWebNavigationDelegate.cs (hunk 2 above)
Comment (jennyf19) [reply core/src/Platforms/iOS/EmbeddedWebview/WKWebNavigationDelegate.cs]: Success is what we've had, so I don't think we should change it. we can discuss off-line though. #Resolved

**Review by henrik-me — APPROVED** [review]
:shipit:

**Pull request merged**